pkg/cmd/cli/sa/gettoken.go

```diff
@@ -0,0 +1,129 @@
+package sa
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/unversioned"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/openshift/origin/pkg/cmd/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/serviceaccounts"
+)
+
+const (
+	GetServiceAccountTokenRecommendedName = "get-token"
+
+	getServiceAccountTokenShort = `Get a token assigned to a service account.`
+
+	getServiceAccountTokenLong = `
+Get a token assigned to a service account.
+
+If the service account has multiple tokens, the first token found will be returned.
+
+Service account API tokens are used by service accounts for privileged access to the API. If actions that cannot
+be accomplished by OpenShift client actions using the service account are required, and the actions necessiate 
+privileged requests to the API server, the service account token is necessary for authorization. This command will
+get the service account token from a service account to facilitate these external actions.
+`
+
+	getServiceAccountTokenUsage = `%s SA-NAME`
+
+	getServiceAccountTokenExamples = `  # Get the service account token from service account 'custom-sa'
+  $ %[1]s 'custom-sa'
+`
+)
+
+type GetServiceAccountTokenOptions struct {
```

Comment:
> The one arg is the name of the service account, and Complete fails if it doesn't exist. What do we need to validate?

Looks like validation is needed for every field.  Code that assumes it is called in the "right" order and thus doesn't need validation is broken code.